src/Dev/Deprecation.php

```diff
@@ -234,6 +237,14 @@ public static function notice($atVersion, $string = '', $scope = Deprecation::SC
             } else {
                 user_error($string, self::$notice_level);
             }
+        } catch (Exception $e) {
+            if ($e instanceof BadMethodCallException && $e->getMessage() === InjectorLoader::NO_MANIFESTS_AVAILABLE) {
+                // noop
+                // this can happen when calling Deprecation::notice() before manifests are available, i.e.
+                // some of the code involved in creating the manifests calls Deprecation::notice()
+            } else {
+                throw $e;
+            }
```

Comment:
I think explicitly catching `BadMethodCallException` here is objectively better than doing a catch all and then checking the instance of exception you got.

My instinct would have been to not have an empty if block ... but I can see how someone might argue this is more readable.

```suggestion
        } catch (BadMethodCallException $e) {
            if ($e->getMessage() === InjectorLoader::NO_MANIFESTS_AVAILABLE) {
                // noop
                // this can happen when calling Deprecation::notice() before manifests are available, i.e.
                // some of the code involved in creating the manifests calls Deprecation::notice()
            } else {
                throw $e;
            }
```

Reply:
Updated